dvc/ui/__init__.py

```diff
@@ -82,17 +82,20 @@ def error_write(
             styled=styled,
         )
 
-    def write_json(
-        self,
-        data: Any,
-        indent: int = 2,
-    ) -> None:
+    def write_json(self, data: Any, indent: int = 2, default=None) -> None:
+        import json
+
         if self.isatty():
+            from rich.highlighter import JSONHighlighter
             from rich.json import JSON
 
-            j = JSON.from_data(data, indent=indent)
-            return self.write(j, styled=True)
-        import json
+            json_instance = JSON.__new__(JSON)
+            j = json.dumps(data, indent=indent, default=default)
+            highlighter = JSONHighlighter()
+            json_instance.text = highlighter(j)
+            json_instance.text.no_wrap = True
+            json_instance.text.overflow = None
+            return self.write(json_instance, styled=True)
```

Comment:
let's just use `highlighter` directly instead of using `JSON` class as we are not getting anything from it anyway.
```suggestion
            j = json.dumps(data, indent=indent, default=default)
            highlighter = JSONHighlighter()
            text = highlighter(j)
            text.no_wrap = True
            text.overflow = None
            return self.write(text, styled=True)
```

Reply:
Sure